iml-manager-cli/src/server.rs

```diff
@@ -218,250 +219,400 @@ async fn wait_till_agent_starts(
     Ok(())
 }
 
-#[allow(clippy::cognitive_complexity)]
-pub async fn server_cli(command: ServerCommand) -> Result<(), ImlManagerCliError> {
-    match command {
-        ServerCommand::List => {
-            let hosts: ApiList<Host> = wrap_fut("Fetching hosts...", get_hosts()).await?;
+fn list_server(hosts: ApiList<Host>) {
+    tracing::debug!("Hosts: {:?}", hosts);
+
+    let table = generate_table(
+        &["Id", "FQDN", "State", "Nids"],
+        hosts.objects.into_iter().map(|h| {
+            vec![
+                h.id.to_string(),
+                h.fqdn,
+                h.state,
+                h.nids.unwrap_or_default().join(" "),
+            ]
+        }),
+    );
+
+    table.printstd();
+}
+
+fn get_profile(config: &AddHosts, profiles: ApiList<ServerProfile>) -> Option<ServerProfile> {
+    let profile_opt = profiles
+        .objects
+        .into_iter()
```

Comment:
I saw some warning about `iter()` vs `into_iter()` with the recent Rust. Not sure if it is the case here.

Reply:
Yes, I saw those as well. Anywhere that recommendation was made I updated accordingly.

Reply:
In my case, if I recall, it suggested `iter()` instead of `into_iter()`. I have no idea :)